src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +87,157 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        return this.resourceStore.requestData().then((data: Object) => {
+            this.removeObsoleteData(this.schema, schema, this.data);
+            const result = this.mergeData(this.schema, schema, this.data, data);
+            this.setMultiple(result);
+
+            this.schema = schema;
+            this.addMissingSchemaProperties();
+            this.validate();
+            this.setSchemaLoading(false);
+        });
     };
 
+    mergeData(
+        oldSchema: Schema,
+        newSchema: Schema,
+        oldData: Object,
+        newData: Object,
+        parentPath: string[] = []
+    ) {
+        let result = {};
+        if (!oldSchema || !newSchema) {
+            return result;
+        }
+
+        for (const key in newSchema) {
```

Comment:
think i would call this `name` to make it different to the `key` used for blocks below